### PR TITLE
Feature/server packages

### PR DESCRIPTION
**Pull request by iOvergaard**
## Description

This re-adds support to load our fake manifests from the server, which should now more or less model the final version, which is currently being built (so this is just a mock).

We are loading raw js files from our mocked App_Plugins folder in anticipation of the real deal from the server.

## Examples

**A custom entrypoint might just log something directly to console, and we see its default exported function being run:**

image

**An extension adds a new section containing a custom element:**

image

**An extension loads a custom property editor UI:**

image

## How to test

Go to `.env.local` (or create it) and put the following in to turn on our mocked server extensions:

```
VITE_UMBRACO_EXTENSION_MOCKS=on
```

### CODE REVIEW

File: src/backoffice/packages/repository/package.repository.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			});
+		});
+	}
+
+	/**
+	 * Request the root items from the Data Source
+	 * @memberOf UmbPackageRepository
+	 * @private
+	 */
+	async #requestRootItems(store: UmbPackageStore) {
```
Comment (madsrasmussen) [inline src/backoffice/packages/repository/package.repository.ts:33]: I think we should make these methods public, at some point, so we can request packages on demand without having to make a new repo instance.

File: src/backoffice/backoffice.element.ts [hunk 2]
```diff
@@ -113,6 +116,9 @@ export class UmbBackofficeElement extends UmbLitElement {
 		this.provideContext(UMB_BACKOFFICE_CONTEXT_TOKEN, new UmbBackofficeContext());
 		this.provideContext(UMB_CURRENT_USER_HISTORY_STORE_CONTEXT_TOKEN, new UmbCurrentUserHistoryStore());
 		new UmbThemeContext(this);
+
+		new UmbPackageStore(this);
+		new UmbServerExtensionController(this, umbExtensionsRegistry);
```
[the remark this answers is not in the document]
Comment (iOvergaard) [reply src/backoffice/backoffice.element.ts:121]: The idea here is that we can reuse this controller in the Auth/Login app and provide it with another instance specific for that app

File: src/backoffice/packages/package-section/views/installed/installed-packages-section-view.element.ts [hunk 3]
```diff
@@ -1,67 +1,57 @@
...
 
-		routes.push({
-			path: '**',
-			redirectTo: 'section/packages/view/installed/overview', //TODO: this should be dynamic
+	/**
+	 * Fetch the installed packages from the server
+	 */
+	private async _loadInstalledPackages() {
+		const package$ = await this.repository.rootItems();
+		package$.subscribe((packages) => {
```
[the remark this answers is not in the document]
Comment (iOvergaard) [reply src/backoffice/packages/package-section/views/installed/installed-packages-section-view.element.ts:32]: Ok good point, we will fix that in Lone's branch where she is expanding this view. Is that ok?

File: src/backoffice/backoffice.element.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (madsrasmussen) [reply src/backoffice/backoffice.element.ts:121]: But wouldn't we have to create a specific instance for this app too then? They are going to talk to the same singleton anyway?

But I will just leave it like this for now, then we can update when we get to the auth app.

**Review by madsrasmussen — APPROVED** [review]

**Pull request merged**